doc: update instructions for openssl updates (#42353)
nodejs/node: doc/contributing/maintaining-openssl.md

This document describes how to update `deps/openssl/`.

If you need to provide updates across all active release lines you will
currently need to generate four PRs as follows:

* a PR for master which is generated following the instructions

>   below for OpenSSL 3.0.x

Reviewer: ```suggestion
  below for OpenSSL 3.0.x.
```

> * a PR for 16.x which is generated following the instructions
>   below for OpenSSL 1.1.1

Reviewer: Shouldn't we use the same wording as other branches?
```suggestion
* a PR for 16.x following the instructions in the v16.x-staging version
  of this guide.
```

Author: That probably makes sense although the 16.x branch has instructions for openssl 3.0 and 1.1.1 as well so we might want to update those at some point.

* a PR for 14.x following the instructions in the v14.x-staging version
  of this guide.

> * a PR which uses the same commit from the second PR to apply the

Author: ```suggestion
* a PR which uses the same commit from the third PR to apply the
```

This document describes how to update `deps/openssl/`.

If you need to provide updates across all active release lines you will
currently need to generate four PRs as follows:

* a PR for master which is generated following the instructions

>   below for OpenSSL 3.0.x.

Reviewer: Wouldn’t we upgrade to semver-minors also?
```suggestion
  below for OpenSSL 3.x.x.
```

    $ git commit openssl
```

> ### OpenSSL 3.0.x

Author: ```suggestion
### OpenSSL 3.x.x
```

% git commit
```

### OpenSSL 3.0.x

```console
    $ make -C deps/openssl/config
    $ git add deps/openssl/config/archs

>     $ git add deps/openssl/openssl

Author: @richard this matches what is in the commit comments below and looks like we have done. It is probably also related to  https://github.com/nodejs/node/issues/42081 as git add deps/openssl/openssl adds all of the headers under `openssl/openssl/include` versus the selective ones that were added for OpenSSL 1.1.1. 

I don't think we want to change right now as we get releases out, but worth looking to see if possibly we can revert to adding just those three, find out why more were needed and possibly ask @danbev when he gets back.

Author: Actually I take that back since I guess the the examples were only showing the `git adds` that were necessary for a particular commit as opposed to what needs to be done in general. Therefore, it likely is not related to https://github.com/nodejs/node/issues/42081 after all.

Reviewer 2: FWIW AFAICT the big difference between OpenSSL 1.1.1 and 3.0.x in Node.js is the `deps/openssl/config/archs` -- there's many more files in OpenSSL 3.0.x under there (see description in https://github.com/nodejs/node/issues/42081)  and a preliminary glance suggested a lot of duplication between the archs.

    $ make -C deps/openssl/config
    $ git add deps/openssl/config/archs
    $ git add deps/openssl/openssl

>     $ git add deps/openssl/config

Author: ```suggestion
```
